### PR TITLE
better error messages for Symbol.asXXX methods

### PR DESCRIPTION
**Pull request by xeno-by**
review by anyone

### CODE REVIEW

File: src/library/scala/reflect/base/Symbols.scala [hunk 1]
```diff
@@ -140,7 +140,17 @@ trait Symbols { self: Universe =>
     /** This symbol cast to a MethodSymbol.
      *  Returns ClassCastException if `isMethod` is false.
```
Comment (adriaanm) [inline src/library/scala/reflect/base/Symbols.scala]: doc needs to be updated

**Comment by adriaanm** [comment]
LGTM provided the doc is updated

**Comment by xeno-by** [comment]
PLS REBUILD ALL

**Comment by xeno-by** [comment]
@jsuereth please, merge

**Pull request merged**